src/PredictionFramework.cpp

```diff
@@ -26,27 +30,89 @@ PredictionFramework::PredictionFramework(
+    (*log_) << "[Fixed parameter values]:\n";
+	(*log_) << "    latent period (theta_l): " << modelInputParameters_.paramlist.T_lat <<std::endl;
+	(*log_) << "    pre-clinical period (theta_i): " << modelInputParameters_.paramlist.T_inf <<std::endl;
+	(*log_) << "    asymptomatic period (theta_r): " << modelInputParameters_.paramlist.T_rec <<std::endl;
+	(*log_) << "    symptomatic period (theta_s): " << modelInputParameters_.paramlist.T_sym <<std::endl;
+	(*log_) << "    hospitalisation stay (theta_h): " << modelInputParameters_.paramlist.T_hos <<std::endl;
+	(*log_) << "    pre-adult probability of symptoms devt (p_s[0]): " << modelInputParameters_.paramlist.juvp_s <<std::endl;
+	(*log_) << "    bed capacity at hospital (K): " << modelInputParameters_.paramlist.K <<std::endl;
+	(*log_) << "    relative infectiousness of asymptomatic (u): " << modelInputParameters_.paramlist.inf_asym <<std::endl;
+
+	//Separate case information for each herd_id
+    modelInputParameters_.seedlist.day_intro = 0;
+    int duration = 0;
+    int time_back;
+    if (modelInputParameters_.seedlist.seedmethod == "background") {
+        time_back = modelInputParameters_.seedlist.hrp;
+    } else {
+        time_back = modelInputParameters_.paramlist.T_inf + modelInputParameters_.paramlist.T_sym;
+
+        if (modelInputParameters_.seedlist.seedmethod != "random") {
+            (*log_) << "Warning!! Unknown seeding method - applying _random_ seed method\n";
+        }
+    }
+
+	const std::vector<int>& regionalCases = observations_.cases[modelInputParameters_.herd_id];
+	const std::vector<int>& regionalDeaths = observations_.deaths[modelInputParameters_.herd_id];
+	const std::vector<int>& timeStamps = observations_.cases[0];
+
+    std::vector<int> obsHosp, obsDeaths;
+    Observations::SelectObservations(duration, modelInputParameters_.seedlist.day_intro,
+        modelInputParameters_.day_shut, obsHosp, obsDeaths, timeStamps, regionalCases,
+        regionalDeaths, time_back, log_);
+
+    (*log_) << "[Health Board settings]:\n";
+	(*log_) << "    SHB id: " << modelInputParameters_.herd_id <<'\n';
+	(*log_) << "    Population size: " << regionalPopulation_ << '\n';
+	(*log_) << "    Number of HCW: " << healthCareWorkers_ << '\n';
+	(*log_) << "    Simulation period: " << duration << "days\n";
+	(*log_) << "    time step: " << modelInputParameters_.tau << "days\n";
+
+    const int n_sim_steps = static_cast<int>(ceil(duration/modelInputParameters_.tau));
+
     Status status = Model::RunModel(
-        parameterSet, fixedParameters_, ageGroupData_, seedlist_, dayShut_, ageNums_,
-        nSimulationSteps, modelStructure_, rng_
+        parameterSet, fixedParameters_, ageGroupData_, modelInputParameters_.seedlist, modelInputParameters_.day_shut, ageNums_,
+        n_sim_steps, modelInputParameters_.model_structure, rng_
     );
 
-    // To do: Write model prediction outputs here
+    double time_taken;
```

Comment:
There's no reason to declare `time_taken` separately from initialising it. It can be changed to `double time_taken = ...`